src/backend/joanie/core/api/client/__init__.py

```diff
@@ -541,6 +542,65 @@ def withdraw(self, request, pk=None):  # pylint: disable=no-self-use, invalid-na
 
         return Response(status=HTTPStatus.NO_CONTENT)
 
+    @extend_schema(
+        request=None,
+        responses={
+            (200, "application/json"): OpenApiTypes.OBJECT,
+            404: serializers.ErrorResponseSerializer,
+            422: serializers.ErrorResponseSerializer,
+        },
+    )
+    @action(detail=True, methods=["POST"])
+    def submit_installment_payment(self, request, pk=None):  # pylint: disable=unused-argument
+        """
+        Pay a failed installment that was scheduled for a given order.
+        """
+        order = self.get_object()
+        if order.state != enums.ORDER_STATE_FAILED_PAYMENT:
+            return Response(
+                {"detail": "The order is not in failed payment state."},
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
+
+        payment_backend = get_payment_backend()
+        installment = order.get_first_installment_refused()
+        if not installment:
+            return Response(
+                {"detail": "No installment found with a refused payment state."},
+                status=HTTPStatus.BAD_REQUEST,
+            )
+
+        credit_card_id = request.data.get("credit_card_id")
+        if not credit_card_id:
+            payment_infos = payment_backend.create_payment(
+                order=order,
+                billing_address=order.main_invoice.recipient_address,
+                installment=installment,
+            )
+            return Response(payment_infos, status=HTTPStatus.OK)
+
+        try:
+            credit_card = CreditCard.objects.get_card_for_owner(
+                pk=credit_card_id, owner_id=order.owner.id
+            )
+        except CreditCard.DoesNotExist:
+            credit_card = False
+
+        if not credit_card:
+            return Response(
+                {"detail": "Credit card does not exist."},
+                status=HTTPStatus.NOT_FOUND,
+            )
```

Comment:
```suggestion
        except CreditCard.DoesNotExist:
            return Response(
                {"detail": "Credit card does not exist."},
                status=HTTPStatus.NOT_FOUND,
            )
```

Reply:
Thanks ! I'll roll back to the first version.